homeworks/14/instagram-api/CommentServlet.java

```diff
@@ -0,0 +1,103 @@
+package com.example.instagram;
+
+import jakarta.servlet.*;
+import jakarta.servlet.http.*;
+import jakarta.servlet.annotation.*;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+@WebServlet(name = "CommentServlet", value = "/posts/*")
+public class CommentServlet extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        response.setContentType("text/html");
+        PrintWriter out = response.getWriter();
+        String pathInfo = request.getPathInfo(); // /{id}/comments
+        String[] pathParts = pathInfo.split("/");
+
+        String method;
+        if(pathParts.length == 2){
+            method = "post";
+        }
+        else {
+            method = "put";
+        }
+        out.println("<html><body>");
+        out.println("<form action=\"posts/*\" method=\""+ method +"\">");
+        out.println("comment: <input type=\"text\" name=\"comment\"><br>");
+        out.println("<input type=\"submit\" value=\"Submit\"></form>");
+        out.println("</body></html>");
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        response.setContentType("text/html");
+        String pathInfo = request.getPathInfo(); // /{id}/comments
+        String[] pathParts = pathInfo.split("/");
+        String post_id = pathParts[1]; // {id}
+//      String part2 = pathParts[2]; // comments
+        String comment = request.getParameter("comment");
+        PrintWriter out = response.getWriter();

```

Comment:
It is good practice to close resources such as PrintWriters when you are finished using them.